### PR TITLE
Change back office user role

### PR DESCRIPTION
**Pull request by irisfaraway**
https://eaflood.atlassian.net/browse/RUBY-761

It should be possible for super users to change the roles of other back office users.

This is dependent on the user group - for example, an agency super can only change the role of other agency users. The roles which are available to choose are also limited by the super user's group - so a finance super can only select finance roles.

### CODE REVIEW

**Comment by irisfaraway** [comment]
Running into failures on the request test because the password isn't validating properly. We got around this with `set` and `update_attribute` previously, but I'd like to see if there is a better solution that still calls our validations.

File: app/controllers/user_activity_controller.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ class UserActivityController < ApplicationController
   include UsersHelper
 
   before_action :assign_user
-  before_action :check_activation_permissions
+  before_action :check_modification_permissions
```
Comment (cintamani) [inline app/controllers/user_activity_controller.rb]: nice one :)

File: app/models/concerns/can_behave_like_user.rb [hunk 2]
```diff
@@ -47,7 +47,6 @@ module CanBehaveLikeUser
     field :locked_at,       type: Time
 
     # Validations
-    validates :password, presence: true, length: { in: 8..128 }
```
Comment (cintamani) [inline app/models/concerns/can_behave_like_user.rb]: Q: Why taking this away?

File: app/models/user.rb [hunk 3]
```diff
@@ -37,21 +37,28 @@ def deactivate!
...
-             finance_super].freeze
+  AGENCY_ROLES = %w[agency
+                    agency_with_refund
+                    agency_super].freeze
+
+  FINANCE_ROLES = %w[finance
+                     finance_admin
+                     finance_super].freeze
+
+  ROLES = (AGENCY_ROLES + FINANCE_ROLES).freeze
```
Comment (cintamani) [inline app/models/user.rb]: 💯

File: app/views/shared/_select_role.html.erb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<% if resource.errors[:role].any? %>
+<div class="form-group form-group-error">
+<% else %>
+<div class="form-group">
+<% end %>
```
Comment (cintamani) [inline app/views/shared/_select_role.html.erb]:
```suggestion
<div class="form-group <%= "form-group-error" if resource.errors[:role].any? %>">
```

File: app/views/shared/_select_role.html.erb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  <fieldset id="role">
+    <legend>
+      <%= t(".roles.legend") %>
+    </legend>
+
+    <% if resource.errors[:role].any? %>
+    <span class="error-message"><%= resource.errors[:role].join(", ") %></span>
```
Comment (cintamani) [inline app/views/shared/_select_role.html.erb]:
```suggestion
      <span class="error-message"><%= resource.errors[:role].join(", ") %></span>
```

File: config/locales/en.yml [hunk 6]
```diff
@@ -34,6 +34,17 @@ en:
       name: "Name"
       position: "Position"
       dob: "Date of birth"
+      shared:
```
Comment (cintamani) [inline config/locales/en.yml]: To delete? :) 
```suggestion
```

Comment (irisfaraway) [reply config/locales/en.yml]: Whoops! Rebase error 🤦‍♀

File: app/models/concerns/can_behave_like_user.rb (hunk 2 above)
Comment (irisfaraway) [reply app/models/concerns/can_behave_like_user.rb]: Did you see my comment on https://github.com/DEFRA/waste-carriers-back-office/pull/569/commits/c0244cfe47869033510724b066ac4bc01b63c712?

**Pull request merged**